main/javascript-modifications/ses-guide.md

```diff
@@ -0,0 +1,506 @@
+until they have been introduced.
+
+To do this, lockdown() tamper-proofs all of the JavaScript intrinsics to prevent 
+prototype pollution. After that, no program can subvert the methods of these objects 
+(preventing some man in the middle attacks). Also, no program can use these mutable
+objects to pass notes to parties that haven't been expressly introduced (preventing 
+some covert communication channels).
+
+For a full explanation of `lockdown()` and its options, please see
+[here](./lockdown.md).
+
+## `harden()`
+
+`harden()` should be called on all objects that will be transferred across a trust boundary 
+The general rule is if you make a new object and give it to someone else (and don't 
+immediately forget it yourself), you should give them `harden(obj)` instead of the raw object. 
+
+After calling lockdown, the harden function ensures that every object in the transitive 
+closure over property and prototype access starting with that object has been frozen by 
+`Object.freeze()`. This means that the object can be passed among programs and none of 
+those programs will be able to tamper with the surface of that object graph. They can 
+only read the surface data and call the surface functions.
+
+You have to harden a class before you harden any of its instances; i.e. it takes two separate 
+steps to harden both a class and its instances. Harden a base class before hardening classes 
+that inherit from it. `harden()` does transitive freezing by following the object’s own
+properties (as opposed to properties it inherited), and the objects whose own properties refer 
+to them, and so forth.
+
+When you freeze an object via `harden()`, it ensures any external callers 
+can only interact with it through functions in the object’s API. `harden()` 
+is an enhanced transitive version of `Object.freeze()`, which only locks up an 
+object's own properties.
+
+All objects that are transferred across a trust boundary must have their API
+surface frozen, usually by calling `harden()`. This ensures other objects can 
+only interact with them through their defined method interface. *CapTP*, our 
+communications layer for passing references to distributed objects, enforces 
+this at vat boundaries.
+
+The general rule is that if you make a new object and give it to someone else 
+(and don't immediately forget it yourself), you should give them `harden(obj)` 
+instead of the raw object. This prevents someone from adding/deleting the properties
+or prototypes of that object. Being hardened doesn't preclude an object from having 
+access to mutable state (`harden(new Map())` still behaves like a normal mutable `Map`), 
+but it means their methods stay the same and can't be surprisingly changed by someone else
+
+Defined objects (`mint`, `issuer`, `zcf`, etc.) shouldn't need hardening as their 
+constructors should do that work. It's mainly records, callbacks, and ephemeral 
+objects that need hardening.
```

Comment:
This could be a breakout using the Agoric environment as an example of uses for `harden`. Would need to rework this to make that clear:
```suggestion
For example, the Agoric programming environment for smart contracts defines objects (`mint`, `issuer`, `zcf`, etc.) that shouldn't need hardening as their 
constructors should do that work. It's mainly records, callbacks, and ephemeral 
objects that need hardening.
```